src/components/game/coreState/utils/FillCheck.js

```diff
@@ -87,17 +93,28 @@ export const checkFilledLines = ({
       }
     }
   }
+  scorekeeper.scoreFilledLines(linesCleared);
 };
 
 // Check all filled targets, remove them from targetBlocks, and erase all
 // covered cells to replace with a call to emptyValue
-export const advanceAndCheckTargets = ({ targets, board, emptyValue }) => {
+export const advanceAndCheckTargets = ({ targets, board, emptyValue, coreState, scorekeeper }) => {
   var gameOver = false;
   targets.forEach((target) => target.advanceUpdate());
   // Clear the targets in a 2nd pass so that the player can hit combos on targets in the same move.
+  var clearedTargets = 0;
   targets.forEach((target) => {
     if (target.isFilled) {
+      clearedTargets += 1;
       target.clear(board, emptyValue);
+      var i = 0;
```

Comment:
First check that the target has a mainCell so we don't set it to undefined for any pieces